src/node_url.cc

```diff
@@ -1698,7 +1699,8 @@ void URL::Parse(const char* input,
         } else {
           if (has_base &&
               base->scheme == "file:") {
-            if (IsNormalizedWindowsDriveLetter(base->path[0])) {
+            if (IsNormalizedWindowsDriveLetter(base->path[0]) &&
+                !(remaining > 0 && IsWindowsDriveLetter(ch, p[1]))) {
```

Comment:
The spec uses the ["starts with a Windows drive letter"](https://url.spec.whatwg.org/#start-with-a-windows-drive-letter) algorithm which checks a few other things. You might want to use that instead by factoring out 

https://github.com/nodejs/node/blob/75606c4f6901ce4086addb0e54b1e88d108eaf8b/src/node_url.cc#L1670-L1676

Reply:
@TimothyGu 👍 something like?

```cpp
static inline bool StartsWithWindowsDriveLetter(char ch, const char* p, int remaining) {
  bool starts_with_drive_letter = false;
  if !IsWindowsDriveLetter(ch, p[1]) || 
      (remaining >= 2 && 
       p[2] != '/' && 
       p[2] != '\\' && 
       p[2] != '?' && 
       p[2] != '#'))) { 
    starts_with_windows_drive_letter = true;
  }
  return starts_with_windows_drive_letter;
}
```

Reply:
whoops logic is reversed in that snippet, but you get the idea; we'd switch the code you shared to !StartsWithDriveLetter().

Reply:
You might want to check `remaining` before accessing `p[1]`?